### PR TITLE
fix:  auth plugin repeats the rewrite phase

### LINKED ISSUE #7276: bug:  runtime error: /usr/local/openresty/lualib/resty/core/request.lua:116: bad argument #1 to 'lower' (string expected, got nil)
### Current Behavior

When use docker apache/apisix:2.13.1-alpine or  apache/apisix:2.14.1-alpine

"GET /test/api/data/page/layout?name=home_v1&platform=app"

请求时报

 [error] 46#46: *8041 lua entry thread aborted: runtime error: /usr/local/openresty/lualib/resty/core/request.lua:116: bad argument #1 to 'lower' (string expected, got nil)

 use docker apache/apisix:2.12.1-alpine 版本正常

### Expected Behavior

_No response_

### Error Logs

_No response_

### Steps to Reproduce

1. Run docker  apache/apisix:2.13.1-alpine or  apache/apisix:2.14.1-alpine
2. GET 正则转发后并且带有get参数的接口时，报500错误
 [error] 46#46: *8041 lua entry thread aborted: runtime error: /usr/local/openresty/lualib/resty/core/request.lua:116: bad argument #1 to 'lower' (string expected, got nil)

### Environment

- APISIX version (run `apisix version`):   apache/apisix:2.13.1-alpine or  apache/apisix:2.14.1-alpine
- Operating system (run `uname -a`): centos 7
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`): /
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):  bitnami/etcd:latest
- APISIX Dashboard version, if relevant: /
- Plugin runner version, for issues related to plugin runners: /
- LuaRocks version, for installation issues (run `luarocks --version`): /

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes #7276

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/plugin.lua [hunk 1]
```diff
@@ -875,14 +875,20 @@ function _M.run_plugin(phase, plugins, api_ctx)
...
     then
         for i = 1, #plugins, 2 do
-            if phase == "rewrite_in_consumer" and plugins[i + 1]._from_consumer
-                    and plugins[i].type ~= "auth"then
+            if ori_phase == "rewrite_in_consumer" and not plugins[i + 1]._from_consumer
+                    and plugins[i].type == "auth" then
+                plugins[i + 1]._skip_rewrite_in_consumer = true
+            end
+
+            if ori_phase == "rewrite_in_consumer" and plugins[i + 1]._from_consumer
```
Comment (spacewander) [inline apisix/plugin.lua:890]: Will it be more readable if we use a local `xxx_phase` variable and keep the `phase` passed as the argument unchanged? The reasons are:
1. better not to rename the variable which is passed as the argument
2. the new `xxx_phase`'s scope will not pass through a step in  `for i = 1, #plugins, 2 do` loop. It won't affect the next plugin.

Comment (tzssangglass) [reply apisix/plugin.lua]: ok, minor refactoring, now the local `xxx_phase` variable is not needed and the `phase` is not modified.

File: apisix/plugin.lua [hunk 2]
```diff
@@ -881,16 +881,21 @@ function _M.run_plugin(phase, plugins, api_ctx)
...
+                phase_func = plugins[i][phase]
             end
-            local phase_func = plugins[i][phase]
 
-            if phase == "rewrite" and plugins[i + 1]._skip_rewrite_in_consumer then
+            if phase == "rewrite_in_consumer" and plugins[i + 1]._skip_rewrite_in_consumer then
                 goto CONTINUE
             end
 
+            ngx.log(ngx.WARN, "phase_func : ", require("inspect")(phase_func))
```
Comment (spacewander) [inline apisix/plugin.lua:898]: let's remove it before merging.

**Review by spacewander — APPROVED** [review]

File: apisix/plugin.lua (hunk 2 above)
Comment (tzssangglass) [reply apisix/plugin.lua]: rm it

**Review by spacewander — DISMISSED** [review]

**Review by leslie-tsang — DISMISSED** [review]

File: t/node/consumer-plugin2.t [hunk 3]
```diff
@@ -238,3 +238,63 @@ x-real-ip: 127.0.0.1
...
+                        },
+                        "ip-restriction": {
+                            "blacklist": [
+                                "127.0.0.0/24"
+                            ]
+                        }
+                    }
+                }]]
+                )
+
```
Comment (bzp2010) [inline t/node/consumer-plugin2.t:265]: Should we check the `code` separately? Two Admin API calls are executed here.

Comment (tzssangglass) [reply t/node/consumer-plugin2.t:270]: update

**Review by bzp2010 — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**